addons/project/static/src/components/project_task_state_selection/project_task_stage_state_selection/project_task_stage_with_state_selection.js

```diff
@@ -0,0 +1,58 @@
+import { _t } from "@web/core/l10n/translation";
+import { standardFieldProps } from "@web/views/fields/standard_field_props";
+import { Many2OneField } from "@web/views/fields/many2one/many2one_field";
+
+import { registry } from "@web/core/registry";
+import { Component } from "@odoo/owl";
+
+import { ProjectTaskStateSelection } from "../project_task_state_selection";
+
+
+export class TaskStageWithStateSelection extends Component {
+    static template = "project.TaskStageWithStateSelection";
+
+    static props = {
+        ...standardFieldProps,
+        stateReadonly: { type: Boolean, optional: true },
+        viewType: { type: String }
+    }
+
+    static components = {
+        ProjectTaskStateSelection,
+        Many2OneField,
+    }
+
+    setup() {
+        this.stateProps = {
+            ...(({ stateReadonly, viewType, ...o }) => o)(this.props),
```

Comment:
This is just to remove these both {stateReadonly, viewType} props from the props we send of the ```ProjectTaskStateSelection``` component(otherwise a error for props validation rises).

Reply:
Updated logic: Here we need to send props to two components ```ProjectTaskStateSelection``` and ```Many2OneField```. Here M20 component doesn't need viewType and stateReadonly(it raises validation if we send extra props). Also ```ProjectTaskStateSelection``` needs viewType prop that why I added back in stateProps also we copy stateReadonly into  readonly to be sent to the component.